### PR TITLE
fix(interactions)

### PR DESCRIPTION
**Pull request by sandymash**
- added place descreption field
- changed the way fields behave
- fixed styles and made map smaller

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/BusinessContactForm/BusinessContactForm.tsx [hunk 1]
```diff
@@ -25,8 +25,7 @@ const BusinessContactForm: React.FC = (): JSX.Element => {
         <div>
             <Typography variant='body1' className={formClasses.fieldName}>פרטי איש קשר:</Typography>
             <Grid container className={formClasses.formRow}>
-                <Grid item xs={3} className={classes.detailsItemField}>
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/BusinessContactForm/BusinessContactForm.tsx:28]: did you remove the class?

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/PlaceNameForm/PlaceNameForm.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import {Grid} from '@material-ui/core';
+import {Controller, useFormContext} from 'react-hook-form';
+import AlphanumericTextField from 'commons/AlphanumericTextField/AlphanumericTextField';
+import FormInput from 'commons/FormInput/FormInput';
+import InteractionEventDialogFields
+    from 'models/enums/InteractionsEventDialogContext/InteractionEventDialogFields';
+import useStyles from './DefaultPlaceEventFormStyles';
+
+const PlaceNameForm = ({nameFieldLabel='שם המוסד'}:Props) => {
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/PlaceNameForm/PlaceNameForm.tsx:10]: move to const

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/PlacesAdditionalForms/HospitalEventForm.tsx [hunk 3]
```diff
@@ -18,24 +18,7 @@ const HospitalEventForm : React.FC = () : JSX.Element => {
     return (
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/PlacesAdditionalForms/HospitalEventForm.tsx:18]: change to medical event form

**Review by RoeiRom — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/BusinessContactForm/BusinessContactForm.tsx (hunk 1 above)
Comment (eladyashar) [reply client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/BusinessContactForm/BusinessContactForm.tsx:28]: now i did, 
there is no use in this styles file anymore

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/InteractionEventForm.tsx [hunk 4]
```diff
@@ -12,13 +12,15 @@ import PlaceSubType from 'models/PlaceSubType';
 import TimePick from 'commons/DatePick/TimePick';
 import FormInput from 'commons/FormInput/FormInput';
 import { get } from 'Utils/auxiliaryFunctions/auxiliaryFunctions';
+import  {getOptionsByPlaceAndSubplaceType} from 'Utils/placeTypesCodesHierarchy';
```
[the remark this answers is not in the document]
Comment (eladyashar) [reply client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/InteractionEventForm.tsx:15]: done

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/InteractionEventForm.tsx [hunk 5]
```diff
@@ -12,13 +12,15 @@ import PlaceSubType from 'models/PlaceSubType';
...
 import InteractionEventDialogData from 'models/Contexts/InteractionEventDialogData';
 import PlacesTypesAndSubTypes from 'commons/Forms/PlacesTypesAndSubTypes/PlacesTypesAndSubTypes';
 import InteractionEventDialogFields from 'models/enums/InteractionsEventDialogContext/InteractionEventDialogFields';
 import InteractionEventContactFields from 'models/enums/InteractionsEventDialogContext/InteractionEventContactFields';
 
-
-import PlaceTypeForm from './PlaceTypeForm';
+import AddressForm from './AddressForm/AddressForm';
+import BusinessContactForm from './BusinessContactForm/BusinessContactForm';
```
[the remark this answers is not in the document]
Comment (eladyashar) [reply client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/InteractionEventForm.tsx:22]: fixed

File: client/src/Utils/placeTypesCodesHierarchy.ts [hunk 6]
```diff
@@ -1,68 +1,145 @@
...
+    const placeConfig = Object.values(placeTypesCodesHierarchy).find(placeTypeConfig => placeTypeConfig.code === placeTypeCode);
+    if(!placeConfig)
+        return defaultOptions;
+    const subPlaceConfig = placeConfig.subTypesCodes &&
+        Object.values(placeConfig.subTypesCodes)
+            .find(subtype => subtype.code === subplaceTypeCode);
+    return  {...placeConfig, ...subPlaceConfig};
+};
+
+type places = 'privateHouse' | 'office' | 'transportation' | 'school' | 'medical' | 'religion' | 'geriatric' | 'otherPublicPlaces';
```
[the remark this answers is not in the document]
Comment (eladyashar) [reply client/src/Utils/placeTypesCodesHierarchy.ts:48]: done

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/PlaceNameForm/PlaceNameForm.tsx (hunk 2 above)
Comment (eladyashar) [reply client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/PlaceNameForm/PlaceNameForm.tsx:10]: done

File: client/src/commons/FormRowWithInput/FormRowWithInput.tsx [hunk 7]
```diff
@@ -9,7 +9,7 @@ interface FormRowWithInputProps {
     appendantLabelIcon?: JSX.Element;
     children: React.ReactElement;
     testId?: string;
-    xs?: GridSize;
+    labelLength?: GridSize;
```
[the remark this answers is not in the document]
Comment (eladyashar) [reply client/src/commons/FormRowWithInput/FormRowWithInput.tsx:12]: done

**Review by RoeiRom — DISMISSED** [review]

**Review by MayaSasson — DISMISSED** [review]

**Review by MayaSasson — APPROVED** [review]

**Review by RoeiRom — APPROVED** [review]

**Pull request merged**